Source/BusinessLogic/Logic/Users/BoardGameGeekUserSaver.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Linq;
+using BusinessLogic.DataAccess;
+using BusinessLogic.Models;
+using BusinessLogic.Models.User;
+using BusinessLogic.Models.Validation;
+
+namespace BusinessLogic.Logic.Users
+{
+    public class BoardGameGeekUserSaver : BaseValidatableRequestSaver, IBoardGameGeekUserSaver
+    {
+        internal const string EXCEPTION_MESSAGE_CURRENT_USER_ALREADY_HAVE_BGG_ACCOUNT_LINKED = "The current user has already a BoardGameGeek account linked.";
+
+        private readonly IDataContext _dataContext;
+
+        public BoardGameGeekUserSaver(IDataContext dataContext)
+        {
+            this._dataContext = dataContext;
+        }
+
+        public BoardGameGeekUserDefinition CreateUserDefintion(CreateBoardGameGeekUserDefinitionRequest request, ApplicationUser currentUser)
+        {
+            ValidateRequest(request);
+            ValidateCurrentUserHasBGGAccount(currentUser);
+
+            var existingItem = _dataContext.GetQueryable<BoardGameGeekUserDefinition>().FirstOrDefault(u => u.Id == request.BoardGameGeekUserId);
```

Comment:
You can probably eliminate the separate request to .Any() and just check to see if you get a result here. No need to query the DB twice (one for .Any() and 2nd time for .FirstOrDefault()).